contents/testing/integration-testing.md

```diff
@@ -0,0 +1,212 @@
+The concept of [Continuous Integration Testing](https://crossbrowsertesting.com/blog/continuous-integration/continuous-integration-testing-delivery/) is also formed to combine integration testing with continuous integration.
+
+:fas-info-circle: More information on instances where integration testing is important can be found on [this article](https://www.guru99.com/integration-testing.html).
+
+## How does integration testing work?
+
+Using this specific system as an example, below are a list of things you should follow to execute effective integration testing:
+
+1. Prepare the integration test plan.
+1. Design the test scenarios, cases and scripts.
+1. Executing the test scripts and report defects if any.
+1. Tracking defects and re-testing application.
+
+### 1. Prepare the integration test plan
+
+Before the actual implementation of your integration tests, it is important to decide on the approach used. There are a few different approaches of integration testing in which you can adopt depending on the development progress of your application.
+
+#### [Big bang approach](https://www.tutorialspoint.com/software_testing_dictionary/big_bang_testing.htm):
+
+This approach involves integrating all the components in your design diagram together and testing everything at once in a complete state. 
+This is convenient but it is difficult to isolate defects and there is a high chance of missing critical underlying defects.
+Big bang integration testing is usually used for smaller applications with few components.
+
+An example of how big bang integration testing can be applied to our given example is shown below:
+
+<center>
+
+![Big bang integration testing diagram](integration-test/big-bang-integration.png "Big bang integration testing diagram]")
+</center>
+
+#### [Incremental testing approach](https://www.quora.com/What-is-incremental-testing-in-software):
+
+This approach involves integrating two or more logically related components. The other related components are added and tested for proper functioning. This is repeated until all the components are joined (or _integrated incrementally_) and tested successfully. It is usually preferred with any application having more than 10 different components.
+
+Incremental integration testing is further split into the 3 approaches shown below:
+
+| | [Top-down approach](https://www.guru99.com/integration-testing.html#9) | [Bottom-up approach](https://www.guru99.com/integration-testing.html#8) | [Hybrid / sandwich approach](https://www.guru99.com/integration-testing.html#10) |
+| -- | -- | -- | -- |
+| **Description**| <tooltip content="Refer to more abstract operations">High level modules</tooltip> are tested first and then <tooltip content="Refer to specific implementation details">lower level modules</tooltip>, until all the modules are tested. | The reverse of top-down approach. | A mix of both top-down bottom-up approaches. |
```

Comment:
I don't think this definition of high-level/low-level is correct, or meaningful to the reader. In this context, high/low level is based on dependency order of components.